### PR TITLE
Attempt at partially supporting symbols

### PR DESCRIPTION
**Pull request by mightyiam**

### CODE REVIEW

File: fixture.js [hunk 1]
```diff
@@ -23,5 +23,8 @@
   NaN: NaN,
   Infinity: Infinity,
   newlines: "foo\nbar\r\nbaz",
-  circular: "[Circular]"
+  circular: "[Circular]",
+  Symbol(): Symbol(),
+  Symbol(foo): Symbol(foo),
```
Comment (mightyiam) [inline fixture.js]: Ideally `[Symbol('foo')]: Symbol('foo')` with quotes according to options.

File: fixture.js [hunk 2]
```diff
@@ -23,5 +23,8 @@
...
+  Symbol(foo): Symbol(foo)
```
Comment (mightyiam) [inline fixture.js]: Ideally
```js
[Symbol.for('foo')]: Symbol.for('foo')
```
 with quotes according to options.

File: index.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ module.exports = (val, opts, pad) => {
 			typeof val === 'number' ||
 			typeof val === 'boolean' ||
 			typeof val === 'function' ||
+			typeof val === 'symbol' ||
```
Comment (mightyiam) [inline index.js]: Handle it!

File: index.js [hunk 4]
```diff
@@ -83,7 +84,7 @@ module.exports = (val, opts, pad) => {
 		}
 
 		if (isObj(val)) {
-			const objKeys = Object.keys(val);
+			const objKeys = Object.keys(val).concat(Object.getOwnPropertySymbols(val));
```
Comment (mightyiam) [inline index.js]: I've just reported [a bug on MDN’s documentation on `Object.keys`](https://bugzilla.mozilla.org/show_bug.cgi?id=1318933).

File: index.js [hunk 5]
```diff
@@ -97,8 +98,8 @@ module.exports = (val, opts, pad) => {
 				}
 
 				const eol = objKeys.length - 1 === i ? tokens.newLine : ',' + tokens.newLineOrSpace;
-				const key = /^[a-z$_][a-z$_0-9]*$/i.test(el) ? el : stringify(el, opts);
-				return tokens.indent + key + ': ' + stringify(val[el], opts, pad + opts.indent) + eol;
+				const key = typeof el === 'symbol' || /^[a-z$_][a-z$_0-9]*$/i.test(el) ? el : stringify(el, opts);
```
Comment (mightyiam) [inline index.js]: Is it correct to bypass the regular expression test when `el` is a symbol?

Comment (SBoudrias) [inline index.js]: Can we make that a if statement. It's getting quite hairy all on a single line.

Comment (mightyiam) [inline index.js]: @SBoudrias how do you like it now?
See last commit.

**Comment by mightyiam** [comment]
@SBoudrias please re- :nerd_face:

**Comment by SBoudrias** [comment]
Just so I understand this properly, this is stripping any symbol in the object?

**Comment by mightyiam** [comment]
@SBoudrias "stripping"? What do you mean by that?

On a different topic, as the PR branch is, non-enumerable symbol-key properties are included. I'm guessing we do not wish for that, correct?

**Comment by mightyiam** [comment]
Now it excludes non-enumerable symbol-key properties.

File: index.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 const isRegexp = require('is-regexp');
 const isObj = require('is-obj');
+const getOwnEnumPropSymbols = require('get-own-enumerable-property-symbols');
```
Comment (mightyiam) [inline index.js]: https://www.npmjs.com/package/get-own-enumerable-property-symbols

File: package.json [hunk 7]
```diff
@@ -29,6 +29,7 @@
     "json"
   ],
   "dependencies": {
+    "get-own-enumerable-property-symbols": "^1.0.1",
```
Comment (mightyiam) [inline package.json]: https://www.npmjs.com/package/get-own-enumerable-property-symbols

**Comment by SBoudrias** [comment]
Cool thanks, the latest diff makes more sense to me now.

**Comment by SBoudrias** [comment]
Cool thanks, the latest diff makes more sense to me now.

**Comment by mightyiam** [comment]
:fireworks: 

I think this is semver-major.

**Comment by SBoudrias** [comment]
I'm going to bump a release tonight.

**Pull request merged**